api/controllers/botupload.go

```diff
@@ -0,0 +1,70 @@
+package controllers
+
+import (
+  "net/http"
+  "github.com/gin-gonic/gin"
+  "github.com/charmbracelet/log"
+  "github.com/Salz-Team/salz/api/models"
+  "strings"
+)
+
+// TODO find a nice home for this
+const (
+  MAX_FILE_SIZE_BYTES = 25 << 20 // ~ 25 MB
+)
+
+func (ctrl *Controller) BotUploadHandler(c *gin.Context) {
+  // TODO get user ID in middleware and pass it around via gin context?
+  authTokenHeader := c.GetHeader("Authorization")
+  authToken := strings.TrimSpace(strings.Replace(authTokenHeader, "Bearer", "", 1))
+
+  // Get the user id from the auth token
+  userToken, err := ctrl.cfg.AuthDBHandler.GetToken(authToken)
+  if err != nil {
+    log.Error("Error getting user id from token", "error", err)
+    c.AbortWithStatus(http.StatusUnauthorized)
+    return
+  }
+  userId := userToken.UserId
+
+  // Note that file is a multipart.File which implements io.Reader
+  file, fileHeader, err := c.Request.FormFile("data")
+  if err != nil {
+    log.Error("Error getting file from form", "error", err)
+    c.AbortWithStatus(http.StatusInternalServerError)
+    return
+  }
+
+  // TODO filesize check, content type, encoding, etc
+
+  // Get the extension of the file
+  extension := fileHeader.Filename[strings.LastIndex(fileHeader.Filename, "."):]
```

Comment:
"Extension" xD Cause `.tar.gz` becomes just `.gz`.

Nothing to do if that's what we intend to do tho.

Reply:
Looking further down, is there a reason why the extension should be extracted here instead of delegating this work to the bot path builder?

Reply:
In the future I figure we probably have a list of extensions that are "allowable" and that we state explicitly map to how the bot gets run. e.g. `.py` files get run with a python interpreter, `.wasm` inside a wasm runtime, etc. 

I feel like the allowable extensions is logic that belongs with the controller rather than the bot itself. Still up for debate, but that's my thought process.

`.tar.gz` vs `.gz`, I don't have a good way of differentiating the extensions there without using everything past the first `.` other than actually reading the file header which is way more effort than I want to do at the moment. And I mean, should `.tar.gz` really be a standalone extension? if its actually a tarball that was gzipped. If you zipped a py file to `bot.py.zip` it feels weird to treat `.py.zip` as its own extension.

Reply:
The bot handler will probably have to do it's own more in-depth parsing (maybe reading header files) anyways.